### PR TITLE
feat: add scaling doc and clean up requirements doc

### PR DESCRIPTION
**Pull request by sharkymark**
PR to:

1. add a vertical and horizontal scaling doc based on recent customer situations
2. clean up the requirements doc, removing specific sizing info or which was a lite version of what is added in the scaling doc

### CODE REVIEW

File: setup/requirements.md [hunk 1]
```diff
@@ -1,43 +1,20 @@
...
-these figures to improve platform performance (we also recommend regular
-performance testing in a staging environment).
+See [Scaling](./scaling.md) for more information.
 
 For **each** active developer using Coder, allocate additional resources. The
 specific amount required per developer varies, though we recommend starting with
-4 CPUs and 16 GB of RAM, then iterating as needed. Developers are free to
-request the resource allocation that fits their usage:
+4 CPUs and 4 GB of RAM, especially when JetBrains IDEs are used and which are
+CPU and memory intensive. Developers are free to request the resource allocation
```
Comment (kylecarbs) [inline setup/requirements.md:16]:
```suggestion
resource intensive. Developers are free to request the resource allocation
```

File: setup/scaling.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+```console
+helm get values coder > values.yaml -n coder
+```
+
+Experiment with increasing CPU and memory requests and limits as the number of
+workspaces in your Coder deployment increase. Pay particular attention to
+whether users have their workspaces configured to auto-start at the same time
+each day, which produces spike loads on the `coderd` pods. To best prevent Out
+of Memory conditions aka OOM Kills, configure the memory requests and limits to
+be the same medibytes (Mi) values. e.g., 8000Mi
```
Comment (kylecarbs) [inline setup/scaling.md:31]:
```suggestion
be the same megabytes (Mi) values. e.g., 8000Mi
```

File: setup/scaling.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+## Database
+
+Coder requires a [PostgreSQL](https://www.postgresql.org) database to store
+metadata related to your deployment.
+
+There is a `timescale` pod meant for evaluations with a CPU and memory resources
+section of the Helm chart that can be configured. However, we recommend
+deploying an external PostgreSQL database on an appropriate machine type to
+handle loads.
```
Comment (kylecarbs) [inline setup/scaling.md:97]: I think we could leave this out, but chefs choice.

**Review by kylecarbs — CHANGES_REQUESTED** [review]

File: setup/scaling.md (hunk 3 above)
Comment (sharkymark) [reply setup/scaling.md]: Good idea. Thanks for your help Kyle.

**Pull request merged**